src/main/java/de/ellpeck/naturesaura/events/CommonEvents.java

```diff
@@ -131,19 +136,25 @@ public void onWorldTick(TickEvent.WorldTickEvent event) {
     @SubscribeEvent
     public void onPlayerTick(TickEvent.PlayerTickEvent event) {
         if (!event.player.world.isRemote && event.phase == TickEvent.Phase.END) {
-            if (event.player.world.getGameTime() % 10 == 0) {
+            if (event.player.world.getGameTime() % 10 == event.player.getEntityId() % 10) {
                 List<ChunkPos> pending = PENDING_AURA_CHUNKS.get(event.player.getUniqueID());
                 pending.removeIf(p -> this.handleChunkWatchDeferred(event.player, p));
             }
 
-            if (event.player.world.getGameTime() % 200 != 0)
+            if (event.player.world.getGameTime() % 200 != event.player.getEntityId() % 200)
+                return;
+
+            ServerPlayerEntity playerMp = (ServerPlayerEntity) event.player;
+            Advancement negativeAdvancement = playerMp.getServerWorld().getServer().getAdvancementManager().getAdvancement(NEGATIVE_ADVANCEMENT_KEY);
+            Advancement positiveAdvancement = playerMp.getServerWorld().getServer().getAdvancementManager().getAdvancement(POSITIVE_ADVANCEMENT_KEY);
+            if(negativeAdvancement != null && positiveAdvancement != null)
```

Comment:
Why would you do this check here? Doesn't this just check if both of the advancements *exist*? The general advancement manager isn't bound to a specific player.

Reply:
You are right

Reply:
I wanna to check if player already have those advancement first
If player already have those advancement, triangulateAuraInArea() may be avoided